### ISSUE TITLE
Gateway UI assumes openfaas-fn is the default namespace

### ISSUE THREAD

**Issue by LucasRoesler** [openfaas/faas#1407]
## My actions before raising this issue
- [ ] Followed the [troubleshooting guide](https://docs.openfaas.com/deployment/troubleshooting/)
- [ ] Read/searched [the docs](https://docs.openfaas.com/)
- [x] Searched [past issues](/issues)

 
If I deploy openfaas and set the default function namespace to something that is not `openfass-fn` the gateway UI will not be able to deploy any functions.  it returns an error

```
unable create Deployment: namespaces "openfaas-fn" not found
```

![image](https://user-images.githubusercontent.com/891889/70986312-f20bf880-20bd-11ea-9707-7bbb7171b264.png)

This error is correct, but not expected. 

## Expected Behaviour

The UI should request the list of namespaces and use the first one in the list as the default.

This of course requires that the api return the configured default as the first value

## Current Behaviour

The ui can not deploy functions unless you explicitly deploy with the `openfaas-fn` namespace.

## Steps to Reproduce (for bugs)

1. Deploy on kubernetes with the latest helm chart `openfaas-5.3.8` but set `namespace=not-default-openfaas` and `functionNamespace=not-default-openfaas-fn`
2. Open the gateway ui
3. Try to deploy `NodeInfo`

## Your Environment

* Are you using Docker Swarm or Kubernetes (FaaS-netes)?
faas-netes

* Operating System and version (e.g. Linux, Windows, MacOS):
osx

**Comment by alexellis** [comment]
Ref: https://github.com/openfaas/faas-netes/issues/511

**Comment by Waterdrips** [comment]
>The UI should request the list of namespaces and use the first one in the list as the default.
@LucasRoesler and @alexellis do we think this is ok? the backend isnt returning my "default" namespace first?

**Comment by LucasRoesler** [comment]
@Waterdrips i think we can update he providers to always put the default namespace first

**Comment by Waterdrips** [comment]
/close

this should have been closed by #1409 but i put the same issue (#1408 ) in the close section twice
